substrate/frame/delegated-staking/src/lib.rs

```diff
@@ -672,50 +674,41 @@ impl<T: Config> Pallet<T> {
 		let mut source_delegation =
 			Delegators::<T>::get(&source_delegator).defensive_ok_or(Error::<T>::BadState)?;
 
-		// some checks that must have already been checked before.
+		// ensure source has enough funds to migrate.
 		ensure!(source_delegation.amount >= amount, Error::<T>::NotEnoughFunds);
 		debug_assert!(
 			!Self::is_delegator(&destination_delegator) && !Self::is_agent(&destination_delegator)
 		);
 
 		let agent = source_delegation.agent.clone();
-		// update delegations
-		Delegation::<T>::new(&agent, amount).update_or_kill(&destination_delegator);
+		// create a new delegation for destination delegator.
+		let _ = Delegation::<T>::new(&agent, amount).update(&destination_delegator);
+		// Provide for the delegator account. This ensures that amount including existential deposit
+		// can be held.
+		frame_system::Pallet::<T>::inc_providers(&destination_delegator);
```

Comment:
wdyt about adding the `inc_providers` within `Delegation::new` for future proofness and make sure we don't forget to increase the providers counter?